src/components/CompareResults/TableHeader.tsx

```diff
@@ -72,19 +77,29 @@ function FilterableColumn({
             theme.palette.mode == 'light'
               ? Colors.Background200
               : Colors.Background200Dark,
-          borderRadius: '4px',
-          fontSize: 'inherit',
+          borderRadius: 0.5,
+          padding: '6px 12px',
         })}
       >
-        {name}
         <FilterListIcon
           fontSize='small'
           color={hasFilteredValues ? 'primary' : 'inherit'}
-          sx={{ marginInlineStart: 1 }}
+          sx={{ marginInlineStart: 1, marginRight: 0.5, marginLeft: 0 }}
           titleAccess={
             hasFilteredValues ? 'Some filters are active' : 'No active filters'
           }
         />
+        {name}
+        <Box
```

Comment:
Note: In the design we don't show the counter when no value are selected, but I think we should still show it in this case because this would filter out every values and give an empty list. So I'm good with this choice.